src/setuptools_scm/__init__.py

```diff
@@ -22,8 +22,8 @@
 # coding: utf-8
 # file generated by setuptools_scm
 # don't change, don't track in version control
-version = {version!r}
-version_tuple = {version_tuple!r}
+version = "{version!s}"
+version_tuple = {version_tuple!s}
```

Comment:
The old variable name will have to keep the old value to prevent breaking users
The new valve will need a new name

Reply:
You mean as far as the `version_tuple`?  The value is not actually a string, it's still a tuple.   I just made the conversion a `s` so that the tuple written contains `""`  instead  of `''`.

Reply:
A template like `short_version ={version_tuple[:2]}` will silently go incorrect with the change of the template variable from tuple to str

Reply:
Hi @RonnyPfannschmidt I apologize, but I honestly don't know what you mean.

The `version_tuple` is still a `tuple` in the printed file.  I just used a `s` conversion in the template.  Note that in the template, `version` is surrounded with quotes, and `version_tuple` is not.

All my code did was change:

```python
# Old
version_tuple = (1, 0, 'dev42')
short_version = {version_tuple[:2]}
print(short_version)   # Prints {(1, 0)}

# New
version_tuple = (1, 0, "dev42")  # All that happens is it went from 'dev42' to "dev42", it's still a tuple though
short_version = {version_tuple[:2]}
print(short_version)   # Prints {(1, 0)}
```

Reply:
as the input is no longer `version_tuple` but `version_tuple_str` its not a tuple one can format as tuple

Reply:
let me re-phase - if the old template before your changes was passed in as template, it should still result-in correct code
currently as the in template type changed from tuple to str it does not

Reply:
Oh, now I understand what you mean.  Yeah lol I was just adding a unit test case and comments to clarify that it's actually a tuple.

Okay, then this becomes harder to solve, because the `!r` conversion in `{version!r}` calls `repr`, which will have the result include single quotes.

Alternate solution:
- Retroactively modify the file written, replacing any `'foo'` with `"foo"`

Maybe this PR makes the `dump_version` code and `test_dump_version` become clunky.

What do you think?  I can implement whatever